### PR TITLE
feat: added home page/context/reducer

### PR DESCRIPTION
**Pull request by aliamaanansari**

### CODE REVIEW

File: src/components/CatItem/styles.jsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  height: 70vh;
+  position: relative;
+`;
+export const Image = styled.img`
+  width: 100%;
+  height: 100%;
+  object-fit: cover;
+  ${mobile({ height: "30vh" })}
+`;
+export const TextHead = styled.div`
```
Comment (vaibhav-mahalle) [inline src/components/CatItem/styles.jsx:19]: like the use of ` for the effective styling of the components

File: src/components/Products/index.jsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+          <Container key={item._id}>
+            <Image src={item.image}></Image>
+            <Link
+              to="/product-list"
+              className="links"
+              onClick={(e) =>
+                dispatch({ type: "SET_CATEGORY", payload: item.categoryName })
+              }
+            >
+              <Heading>
```
Comment (vaibhav-mahalle) [inline src/components/Products/index.jsx:26]: very well broken code into the components

File: src/components/Slider/index.jsx [hunk 3]
```diff
@@ -0,0 +1,187 @@
+import React, { useContext } from "react";
+import { MainContext } from "../../Context";
+import {
+  Category,
```
Comment (vaibhav-mahalle) [inline src/components/Slider/index.jsx:4]: well-formatted and clean code.

**Review by vaibhav-mahalle — APPROVED** [review]
learnt a clever use of ` for styling of the components

File: src/components/CatItem/styles.jsx (hunk 1 above)
Comment (aliamaanansari) [reply src/components/CatItem/styles.jsx:19]: thanks

File: src/components/Slider/index.jsx (hunk 3 above)
Comment (aliamaanansari) [reply src/components/Slider/index.jsx:4]: thanks

File: src/components/Products/index.jsx (hunk 2 above)
Comment (aliamaanansari) [reply src/components/Products/index.jsx:26]: thanks

**Pull request merged**